pkg/planner/core/indexmerge_path.go

```diff
@@ -478,6 +595,159 @@ func (ds *DataSource) generateIndexMergeAndPaths(normalPathCnt int) *util.Access
+	usedMap := make([]bool, len(indexMergeDNFConds))
+	needSelection := false
+
+	for offset, dnfCond := range indexMergeDNFConds {
+		var cnfConds []expression.Expression
+		if sf, ok := dnfCond.(*expression.ScalarFunction); !ok {
+			continue
+		} else if sf.FuncName.L == ast.LogicAnd {
+			cnfConds = expression.FlattenCNFConditions(sf)
+		} else {
+			cnfConds = []expression.Expression{sf}
+		}
+		// for every dnf condition, find the most suitable mv index path.
+		// otherwise, for table(a json, b json, c int, idx(c,a), idx2(b,c))
+		// condition: (1 member of (a) and c=1 and d=2) or (2 member of (b) and c=3 and d=2);
+		// will both pick(c,a) idx with range [1 1, 1 1] and [3,3], the latter can pick the most
+		// valuable index idx2 with range [2 3,2 3]
+		var (
+			bestPaths            []*util.AccessPath
+			bestCountAfterAccess float64
+			bestNeedSelection    bool
+		)
+		for idx := 0; idx < len(possibleMVIndexPaths); idx++ {
+			idxCols, ok := ds.prepareCols4MVIndex(possibleMVIndexPaths[idx].Index)
+			if !ok {
+				continue
+			}
+			// for every cnfCond, try to map it into possible mv index path.
+			// remainingFilters is not cared here, because it will be all suspended on the table side.
+			accessFilters, remainingFilters := ds.collectFilters4MVIndex(cnfConds, idxCols)
+			if len(accessFilters) == 0 {
+				continue
+			}
+			paths, isIntersection, ok, err := ds.buildPartialPaths4MVIndex(accessFilters, idxCols, possibleMVIndexPaths[idx].Index)
+			if err != nil {
+				logutil.BgLogger().Debug("build index merge partial mv index paths failed", zap.Error(err))
+				return nil, nil, false, err
+			}
+			if isIntersection || !ok { // limitation 2
+				continue
+			}
+			if len(paths) == 0 {
+				continue
+			}
+			// UNION case, use the max count after access for simplicity.
+			maxCountAfterAccess := -1.0
+			for _, p := range paths {
+				maxCountAfterAccess = math.Max(maxCountAfterAccess, p.CountAfterAccess)
+			}
+			// Note that: here every path is about mv index path.
+			// find the most valuable mv index path, which means it has the minimum countAfterAccess.
+			if len(bestPaths) == 0 {
+				bestPaths = paths
+				bestCountAfterAccess = maxCountAfterAccess
+				bestNeedSelection = len(remainingFilters) != 0
+			} else {
+				if bestCountAfterAccess > maxCountAfterAccess {
+					bestPaths = paths
+					bestCountAfterAccess = maxCountAfterAccess
+					bestNeedSelection = len(remainingFilters) != 0
+				}
+			}
+		}
+		if len(bestPaths) != 0 {
+			usedMap[offset] = true
+			// correctly find a dnf condition for this mv index path
+			mvAndPartialPaths = append(mvAndPartialPaths, bestPaths...)
+			if !needSelection && bestNeedSelection {
+				// collect one path's need selection flag.
+				needSelection = bestNeedSelection
+			}
+		}
+	}
+	return mvAndPartialPaths, usedMap, needSelection, nil
+}
+
+func (ds *DataSource) generateIndexMergePartialPaths4And(normalPathCnt int,
```

Comment:
Please add some comments for this function, especially a concrete example like "the input is ..., the output is ..."